rosserial_client/src/ros_lib/ros/node_handle.h

```diff
@@ -200,8 +200,17 @@ namespace ros {
         while( true )
         {
           int data = hardware_.read();
-          if( data < 0 )
+          if( data < 0 ) {
+#if defined(USE_WIFICON)
+		    if (!hardware_.isConnectionUp()) {		// check connection, if it is broken try to repair it N.B. should really add interrupt timer as this blocks !
+				configured_ = false;
+				return -3;
+			} else
+				break;
+#else
```

Comment:
The client is abstract— all it worries about is sending and receiving characters. It manages for itself whether it is "connected" based on its time sync pings which go back and forth. So my objection is simply that there should be another way to manage this than adding more logic to the client.

The place for connection-managing logic is the Hardware class, IMO— you can either check it in the `read` method, or have something else which periodically checks the connection and attempts to recreate it as necessary.